### PR TITLE
[Feature] Add customer API preset to codegen

### PR DESCRIPTION
**Pull request by patlai**
### WHY are these changes introduced?
With the introduction of the customer API in dev preview, Shopify apps, extensions and headless storefronts will be developed using the customer API

### WHAT is this pull request doing?
Adds the customer API as a codegen preset and add logic to the helpers that allow generating types for it

## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: packages/api-clients/api-codegen-preset/README.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ Use this function if you want to configure a custom project, or add your own `ge
 | ------------ | ----------- | -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
 | apiType      | `ApiType`   | _N/A_                | The API to pull schemas from.                                                                                                                                      |
 | apiVersion   | `string?`   | Latest               | Pull schemas for a specific version.                                                                                                                               |
+| apiKey       | `string?`   | _N/A_                | The app's API key. This can be found in the app's page in Shopify Partners in the "Authentication" section. Required for the customer API preset and helpers       |
```
Comment (paulomarg) [inline packages/api-clients/api-codegen-preset/README.md:100]: Should we say "only valid for the customer API" for now?

File: packages/api-clients/api-codegen-preset/src/helpers/get-schema-data.ts [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
 export function getSchemaData(
   outputDir: string,
   apiType: ApiType,
   apiVersion?: string,
+  apiKey?: string,
 ) {
   const config = apiConfigs[apiType];
 
+  if (apiType === ApiType.Customer) {
+    if (!apiKey) {
```
Comment (paulomarg) [inline packages/api-clients/api-codegen-preset/src/helpers/get-schema-data.ts:14]: Should we also require `apiVersion` here?

File: packages/api-clients/api-codegen-preset/src/helpers/get-schema-data.ts [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+      throw new Error('The customer API requires an API key');
+    }
+
+    const schemaUrl = (
+      apiVersion
+        ? config.schema.replace('%%API_VERSION%%', apiVersion)
+        : config.schema.replace('&api_version=%%API_VERSION%%', '')
+    ).replace('%%API_KEY%%', apiKey);
```
Comment (paulomarg) [inline packages/api-clients/api-codegen-preset/src/helpers/get-schema-data.ts:22]: Took me a second to get this - could we replace each of the values in a separate step to make it more readable?

File: packages/api-clients/api-codegen-preset/src/tests/api-project.test.ts [hunk 4]
```diff
@@ -28,7 +29,17 @@ describe('shopifyApiProject', () => {
 
         // THEN
         expect(projectConfig).toEqual({
-          schema: `https://shopify.dev/${type}-graphql-direct-proxy/2023-10`,
+          schema:
```
Comment (paulomarg) [inline packages/api-clients/api-codegen-preset/src/tests/api-project.test.ts:32]: There's a little bit of repetition here - I wonder if we could pull this into a private method?

**Review by paulomarg — COMMENTED** [review]
Just a couple of nits, but looking good!

File: packages/api-clients/api-codegen-preset/src/tests/api-project.test.ts (hunk 4 above)
Comment (patlai) [reply packages/api-clients/api-codegen-preset/src/tests/api-project.test.ts]: yup, added a helper

**Review by paulomarg — APPROVED** [review]
LGTM, thank you!

**Pull request merged**